### PR TITLE
frontend/fix: #213 fixed contacts bug

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs [hunk 1]
```diff
@@ -222,11 +222,11 @@ contactsValidation contacts = do
   let newContacts = map (\x ->
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs:222]: Can we create a better function for validation without these many if else

Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs:222]: @chakri642 you can use `replaceAll` also to reduce number of checks

File: Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs [hunk 2]
```diff
@@ -100,36 +100,63 @@ eval (PopUpModalAction PopUpModal.OnButton2Click) state = do
...
+      ]
+  else do
+    let newContacts = getContactList allContacts
+    localContacts <- pure $ getValueToLocalStore CONTACTS
+    contactsInJson <- pure $ parseNewContacts localContacts
+    let validContacts = (map (\contact -> 
+      if ((DS.length contact.number) > 10 && (DS.length contact.number) <= 12 && ((DS.take 1 contact.number) == "0" || (DS.take 2 contact.number) == "91")) then
+        contact {number =  DS.drop ((DS.length contact.number) - 10) contact.number}
+      else contact 
+    ) newContacts)
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs:133]: remove `validContacts` since we are not using it

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs [hunk 3]
```diff
@@ -100,36 +100,63 @@ eval (PopUpModalAction PopUpModal.OnButton2Click) state = do
   let newContacts = filter (\x -> x.number <> x.name /= state.data.removedContactDetail.number <> state.data.removedContactDetail.name) state.data.contactsList
   contactsInString <- pure $ toString newContacts
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/CustomerUtils/EmergencyContactsScreen/Controller.purs:101]: you can use it like `let contactsInString = toString newContacts`

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**